src/hotspot/share/classfile/classLoaderData.cpp

```diff
@@ -1034,6 +1051,19 @@ void ClassLoaderData::verify() {
   if (_modules != NULL) {
     _modules->verify();
   }
+
+  if (_deallocate_list != nullptr) {
+    for (int i = _deallocate_list->length() - 1; i >= 0; i--) {
+      Metadata* m = _deallocate_list->at(i);
+      if (m->is_klass()) {
+        ((InstanceKlass*)m)->verify();
```

Comment:
There's no `InstanceKlass::verify()` method. Do you mean  `((Klass*)m)->verify()` ?

Reply:
// Verification
  virtual void verify_on(outputStream* st);
  void verify() { verify_on(tty); }

It calls the virtual function InstanceKlass::verify_on()

Reply:
Yes, I did add the deallocate_list verification too.  I can change the title of the bug.

Reply:
This code seem unsafe since you are checking for `is_klass()` but cast to `InstanceKlass*`. Since `verify()` is in `Klass` anyway, it's better to say

```
 if (m->is_klass()) {
   ((Klass*)m)->verify();
```

Reply:
fixed.